_posts/2020/2020-03-01-fix-first-last-explicitly-set-implicit-ordering-in-rails-with-uuids.md

```diff
@@ -0,0 +1,49 @@
+---
+title: "‘Fix’ first & last by explicitly setting implicit ordering"
+description: "When UUIDs and Rails make a mess"
+layout: article
+category: ruby
+image:
+  base: '2020/fix-first-last-explicitly-set-implicit-ordering-in-rails-with-uuids'
+  alt: "Pins on a red mat"
+  credit:  Jeff Frenette
+  source: "https://unsplash.com/photos/Y_AWfh0kGT4"
+---
+
+I strongly suggest for new applications, to [use UUIDs as primary keys](/ruby/choose-uuids-for-model-ids-in-rails). However this causes issues in the implicit ordering that Rails performs.
+
+I proposed a method for rectifying that by [using named scopes alongside `first` and `last`](/ruby/first-and-last-may-not-mean-what-you-think), but since that article there is an easier way to re-enable the default behaviour of Active Record.
```

Comment:
```suggestion
In a previous article, I suggested [using named scopes alongside `first` and `last`](/ruby/first-and-last-may-not-mean-what-you-think), but there is now an easier way to re-enable the default behaviour of Active Record.
```

Reply:
Oh — really? Exciting!! Since I've been UUIDs myself since that article! I'm excited to read on. 😁